resharper/src/resharper-unity/CSharp/Daemon/Errors/CSharpErrors.xml

```diff
@@ -180,6 +204,53 @@
     <Range>EqualityExpression.GetHighlightingRange()</Range>
     <Behavour overlapResolvePolicy="WARNING" />
   </Warning>
+  
+  <Warning name="UseExplicitTypeInsteadOfStringUsing" configurableSeverity="Unity.UseExplicitTypeInsteadOfStringUsing">
+    <Parameter type="IInvocationExpression" name="invocationMethod" />
+    <Parameter type="string" name="methodName" isValid="skip"/>
+    <Parameter type="string" name="stringLiteral" isValid="skip"/>
+    <Parameter type="IExpression" name="argument" />
+    <Parameter type="ITypeElement[]" name="availableTypes" isValid="skip"/>
+    <Message value="Using the string for the component type variants results in additional overhead and is not safe. 
+    Suggestion is transform from '.{0}(\&quot;{1}\&quot;)' to '.{0}&lt;{1}&gt;()'.">
```

Comment:
"Prefer generic method overload". If you really want to add the method: "Prefer generic method overload 'GetComponent<Foo>()'"